src/conversationLogger.js

```diff
@@ -1,92 +1,99 @@
-      text = meta.attachment.payload.url;
-    } else {
-      text = `Unknown meta.attachment: \`${JSON.stringify(meta.attachment)}\``;
+    if (this.options.slackWebhookUrl && this.options.slackChannel) {
+      this.logger.add(Slack, {
+        webhook_url: this.options.slackWebhookUrl,
+        username: 'Chat Spy',
+        custom_formatter: this.slackFormatter
+      });
     }
-  } else if (meta.attachments) {
-    text = meta.attachments.map((attachment) => {
-      if (attachment.payload && attachment.payload.url) {
-        return attachment.payload.url;
-      } else {
-        return 'Unknown meta.attachments[]: `' + JSON.stringify(attachment) + '`';
-      }
-    }).join('\n');
   }
 
-  return {
-    channel: process.env.SLACK_CHANNEL,
-    icon_url: `https://robohash.org/${meta.userId}.png`,
-    username: meta.userId,
-    text: addressee + text
-  };
-}
+  slackFormatter(level/*: string */, msg/*: Object */, meta/*: Object */) {
+    // console.log(JSON.stringify(meta));  // enable for DEBUG
+    const addressee = meta.userId === meta.recipientId ? '' : '> ';
+    let text = '```' + JSON.stringify(meta, undefined, 2) + '\n```';
+    if (meta.text) {
+      text = meta.text;
+    } else if (meta.payload) {
+      text = '`' + meta.payload + '`';
+    } else if (meta.attachment) {
+      if (meta.attachment.type === 'template') {
+        text = meta.attachment.payload.elements.map((element) => element.title).join('\n');
+      } else if (meta.attachment.payload.url) {
+        text = meta.attachment.payload.url;
+      } else {
+        text = `Unknown meta.attachment: \`${JSON.stringify(meta.attachment)}\``;
+      }
+    } else if (meta.attachments) {
+      text = meta.attachments.map((attachment) => {
+        if (attachment.payload && attachment.payload.url) {
+          return attachment.payload.url;
+        } else {
+          return 'Unknown meta.attachments[]: `' + JSON.stringify(attachment) + '`';
+        }
+      }).join('\n');
+    }
 
-if (process.env.SLACK_WEBHOOK_URL && process.env.SLACK_CHANNEL) {
-  logger.add(Slack, {
-    webhook_url: process.env.SLACK_WEBHOOK_URL,
-    username: 'Chat Spy',
-    custom_formatter: slackFormatter
-  });
-}
+    return {
+      channel: this.options.slackChannel,
+      icon_url: `https://robohash.org/${meta.userId}.png`,
+      username: meta.userId,
+      text: addressee + text
+    };
+  }
 
+  logIncoming(requestBody/*: Object */) {
+    if (this.dashbotClient) {
+      this.dashbotClient.logIncoming(requestBody);
+    }
+    const data = requestBody.entry[0].messaging[0];
+    if (!(data.postback || data.message)) {
+      // console.log(JSON.stringify(data));  // enable for DEBUG
+      return;
+    }
 
-function logIncoming(requestBody) {
-  if (dashbotClient) {
-    dashbotClient.logIncoming(requestBody);
+    this.logger.info(Object.assign({
+      userId: data.sender.id,
+      senderId: data.sender.id
+    }, data.message || {}, data.postback || {}));
```

Comment:
this was kind of a mistake. I originally only had to deal with messages, and then postbacks came in and I just shoved 'em in. A rewrite is due.